snapcraft/internal/repo/_deb.py

```diff
@@ -122,8 +122,11 @@ def _setup_apt(self, cache_dir):
                 "Cannot find 'dpkg' command needed to support multiarch")
 
         apt_cache = apt.Cache(rootdir=cache_dir, memonly=True)
-        apt_cache.update(fetch_progress=self.progress,
-                         sources_list=sources_list_file)
+        try:
+            apt_cache.update(fetch_progress=self.progress,
+                             sources_list=sources_list_file)
+        except apt.cache.FetchFailedException:
```

Comment:
what is the current exception message? We should consider printing it.

Reply:
We can't assume that there is a message. If there is one, like in the linked bug, the message might look like so:

E:Failed to fetch copy:/home/alan/.cache/snapcraft/stage-packages/apt/a810ed86fa085532f20ab86eb7554a1ff527e7f8f46e49c343abc3d7c4d141ae0c39e374361c05f4561cf2aec6689aac/var/lib/apt/lists/partial/security.ubuntu.com_ubuntu_dists_xenial-security_multiverse_dep11_icons-64x64.tar.gz Hash Sum mismatch, E:Failed to fetch copy:/home/alan/.cache/snapcraft/stage-packages/apt/a810ed86fa085532f20ab86eb7554a1ff527e7f8f46e49c343abc3d7c4d141ae0c39e374361c05f4561cf2aec6689aac/var/lib/apt/lists/partial/gb.archive.ubuntu.com_ubuntu_dists_xenial-backports_main_dep11_icons-64x64.tar.gz Hash Sum mismatch, E:Some index files failed to download. They have been ignored, or old ones used instead.

To my mind this doesn't seem very user-friendly. In the same situation, the printed output before that will have `Err http://security.ubuntu.com/ubuntu xenial-security/multiverse DEP-11 128x128 Icons
  Hash Sum mismatch` at the end, so I'd rather be looking at that.